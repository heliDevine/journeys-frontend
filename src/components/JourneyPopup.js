import PropTypes from 'prop-types';

const JourneyPopup = ({ selectedJourney, onClose }) => {
  return (
    <div className="popup-container">
      <div className="popup-content">
        <h2>Change this title</h2>
        <p>
          {(selectedJourney.distance / 1000).toFixed(1)} kilometers and lasted{' '}
          {Math.floor(selectedJourney.duration / 60)} minutes and{' '}
          {selectedJourney.duration % 60} seconds
        </p>
        <p>
          CO2 emissions saved by taking this journey by bike was{' '}
          {((selectedJourney.distance / 1000) * 107.5).toFixed(0)}gCO2
        </p>
        <button className="popup-close-button" onClick={onClose}>
          Close
        </button>
      </div>
    </div>
  );
};

export default JourneyPopup;

JourneyPopup.propTypes = {
  selectedJourney: PropTypes.shape({
    distance: PropTypes.number.isRequired,
    duration: PropTypes.number.isRequired,
  }),
  onClose: PropTypes.func.isRequired,
};
