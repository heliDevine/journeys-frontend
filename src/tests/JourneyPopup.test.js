import { render, screen } from '@testing-library/react';
import JourneyPopup from '.././components/JourneyPopup';

describe('JourneyPopup', () => {
  const validProps = {
    selectedJourney: {
      departureStationName: 'Test departure station',
      returnStationName: 'Test return station',
      distance: 1000,
      duration: 3000,
    },
    onClose: () => {},
  };
  it('renders correctly', () => {
    const { asFragment } = render(
      <JourneyPopup
        selectedJourney={validProps.selectedJourney}
        onClose={validProps.onClose}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Button renders with correct text', () => {
    render(
      <JourneyPopup
        selectedJourney={validProps.selectedJourney}
        onClose={validProps.onClose}
      />,
    );
    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toBeInTheDocument();
  });
});
