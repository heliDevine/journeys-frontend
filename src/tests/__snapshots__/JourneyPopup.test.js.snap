// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JourneyPopup renders correctly 1`] = `
<DocumentFragment>
  <div
    class="popup-container"
  >
    <div
      class="popup-content"
    >
      <h2>
        Change this title
      </h2>
      <p>
        1.0 kilometers and lasted 50 minutes and 0 seconds
      </p>
      <p>
        CO2 emissions saved by taking this journey by bike was 108gCO2
      </p>
      <button
        class="popup-close-button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
